src/sentry/snuba/sessions_v2.py

```diff
@@ -283,7 +283,7 @@ def __init__(self, query, params, allowed_resolution: AllowedResolution):
         query_columns = set()
         for i, field in enumerate(self.fields.values()):
             columns = field.get_snuba_columns(raw_groupby)
-            if i == 0:
+            if i == 0 or field == "sum(session)":  # Prefer first, but sum(session) always wins
```

Comment:
This is needed to keep consistency between how the two implementations (sessions and metrics) sort the results (see test case below). The previous metrics-based implementation simply ran another snuba query when the first one returned an empty result, but we cannot do that with the current implementation because we now only make a single `MetricsQuery`. https://github.com/getsentry/sentry/blob/7e1e3a904c0cc3a2b0ace7e7df7eb8bc9b4c7498/tests/snuba/api/endpoints/test_organization_sessions.py#L882-L885.